### ISSUE TITLE
⌨ Profile page - "On this page" text should be a header.

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#4997]
## ❌ Current behaviour (issue)
The “On this page” text before the menu is not coded as a header.
<img width="302" alt="Screen Shot 2022-11-29 at 1 11 24 PM" src="https://user-images.githubusercontent.com/1794660/205375208-cb1950b1-0e7c-43c0-94bb-ab081c75c8a2.png">

**WCAG 2.1 Criteria:** 1.3.1 - Info & Relationships, 2.4.6 - Headings and Labels

### How to reproduce
1. Environment (prod or UA?) Prod: talent.canada.ca/en/users/.../profile
2. Which browser and/or Assistive Technology? Chrome

## ✔ Expected behaviour (fix)
The “On this page” text before the menu should be a heading lvl 2.

### Acceptance Criteria
A set of assumptions which, when tested, verify that the accessibility requirement was met.
- [ ]
- [ ] 

## Additional information (optional)
1. How does it impact users?
2. Related issues, suggestions how to fix: Code as an `<h2>`
